e2e/svc_mn_1786387615.py

```
@@ -0,0 +1,19 @@
+
+
+def find_user(conn: sqlite3.Connection, username: str):
+    # SQL injection: the username is concatenated straight into the statement.
+    cursor = conn.cursor()
+    cursor.execute("SELECT id, email FROM users WHERE name = '" + username + "'")
+    return cursor.fetchone()
+
+
+def make_session_token(user_id: str) -> str:
+    # MD5 is not a suitable hash for session tokens.
+    return hashlib.md5(user_id.encode()).hexdigest()
```

Ruff (0.16.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)